pkg/controllers/nodeclaim/disruption/underutilization_test.go

```diff
@@ -0,0 +1,272 @@
+		fmt.Printf("this is the fake clock time: %s\n this is the apiserver time %s", fakeClock.Now(), nodeClaim.CreationTimestamp.Time)
+		// set the fake clock to the same time as the creation timestamp of the nodeclaim.
+		// this is necessary to avoid differences in the api server clock vs our injected clock
+		fakeClock.SetTime(nodeClaim.CreationTimestamp.Time)
+
+		ExpectMakeNodesAndNodeClaimsInitializedAndStateUpdated(ctx, env.Client, nodeStateController, nodeClaimStateController, []*corev1.Node{node}, []*v1.NodeClaim{nodeClaim})
+		fakeClock.Step(30 * time.Second)
+		ExpectObjectReconciled(ctx, env.Client, nodeClaimDisruptionController, nodeClaim)
+
+		nodeClaim = ExpectExists(ctx, env.Client, nodeClaim)
+		Expect(nodeClaim.StatusConditions().Get(v1.ConditionTypeUnderutilized).IsTrue()).To(BeFalse())
+		fakeClock.Step(30 * time.Second)
+
+		timeNow := fakeClock.Now()
+		fmt.Printf("this is the time now %s\n", timeNow)
+		ExpectObjectReconciled(ctx, env.Client, nodeClaimDisruptionController, nodeClaim)
+
+		nodeClaim = ExpectExists(ctx, env.Client, nodeClaim)
+		Expect(nodeClaim.StatusConditions().Get(v1.ConditionTypeUnderutilized).IsTrue()).To(BeTrue())
+	})
+	It("should mark NodeClaims as underutilized based on cluster state initialization", func() {
+		ExpectApplied(ctx, env.Client, nodePool, nodeClaim, node)
+		// set the fake clock to the same time as the creation timestamp of the nodeclaim.
+		// this is necessary to avoid differences in the api server clock vs our injected clock
+		fakeClock.SetTime(nodeClaim.CreationTimestamp.Time)
+
+		fakeClock.Step(30 * time.Second)
+		ExpectMakeNodesAndNodeClaimsInitializedAndStateUpdated(ctx, env.Client, nodeStateController, nodeClaimStateController, []*corev1.Node{node}, []*v1.NodeClaim{nodeClaim})
+		ExpectObjectReconciled(ctx, env.Client, nodeClaimDisruptionController, nodeClaim)
+
+		// reset the cluster state
+		cluster.SetInitTime(fakeClock.Now())
+		fakeClock.Step(30 * time.Second)
+		ExpectMakeNodesAndNodeClaimsInitializedAndStateUpdated(ctx, env.Client, nodeStateController, nodeClaimStateController, []*corev1.Node{node}, []*v1.NodeClaim{nodeClaim})
+		ExpectObjectReconciled(ctx, env.Client, nodeClaimDisruptionController, nodeClaim)
+
+		Expect(nodeClaim.StatusConditions().Get(v1.ConditionTypeUnderutilized).IsTrue()).To(BeFalse())
+		ExpectObjectReconciled(ctx, env.Client, nodeClaimDisruptionController, nodeClaim)
+
+		fakeClock.Step(30 * time.Second)
+		ExpectMakeNodesAndNodeClaimsInitializedAndStateUpdated(ctx, env.Client, nodeStateController, nodeClaimStateController, []*corev1.Node{node}, []*v1.NodeClaim{nodeClaim})
+		ExpectObjectReconciled(ctx, env.Client, nodeClaimDisruptionController, nodeClaim)
+
+		nodeClaim = ExpectExists(ctx, env.Client, nodeClaim)
+		Expect(nodeClaim.StatusConditions().Get(v1.ConditionTypeUnderutilized).IsTrue()).To(BeTrue())
+	})
+	It("should clear the underutilized condition when a pod is added", func() {
```

Comment:
Do we have a test that adds a pods and then ensures that the underutilized condition isn't removed until we have fully exceeded the consolidateAfter timeout?